bot/connector-open-ai/src/main/kotlin/OpenAIConnectorCallback.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2017/2025 SNCF Connect & Tech
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.bot.connector.openai
+
+import ai.tock.bot.connector.ConnectorCallbackBase
+import ai.tock.bot.engine.action.Action
+import ai.tock.bot.engine.action.SendSentence
+import ai.tock.bot.engine.event.MetadataEvent
+import ai.tock.bot.engine.event.hasStreamMetadata
+import ai.tock.shared.serialization.writeJson
+import ai.tock.shared.vertx.sendSseEnd
+import ai.tock.shared.vertx.sendSseMessage
+import io.vertx.core.http.HttpHeaders
+import io.vertx.core.http.HttpServerResponse
+import io.vertx.ext.web.RoutingContext
+import java.util.Locale
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.CopyOnWriteArrayList
+
+class OpenAIConnectorCallback(
+    applicationId: String,
+    val locale: Locale,
+    private val context: RoutingContext?,
+    private val actions: MutableList<Action> = CopyOnWriteArrayList(),
+    private val metadata: MutableMap<String, String> = ConcurrentHashMap(),
```

Comment:
Is this metadata actually used anywhere?

Reply:
yes, for streaming